WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/FeeOrDiscountLineDetailsViewModelTests.swift

```diff
@@ -375,18 +394,36 @@ final class FeeOrDiscountLineDetailsViewModelTests: XCTestCase {
         // Given
         let analytics = MockAnalyticsProvider()
         let viewModel = FeeOrDiscountLineDetailsViewModel(isExistingLine: false,
-                                                baseAmountForPercentage: 0,
-                                                initialTotal: "",
-                                                lineType: .discount,
-                                                locale: usLocale,
-                                                storeCurrencySettings: usStoreSettings,
+                                                          baseAmountForPercentage: 0,
+                                                          initialTotal: .zero,
+                                                          lineType: .discount,
+                                                          locale: usLocale,
+                                                          storeCurrencySettings: usStoreSettings,
                                                           analytics: WooAnalytics(analyticsProvider: analytics),
-                                                didSelectSave: { _ in })
+                                                          didSelectSave: { _ in })
 
         // When
         viewModel.removeValue()
 
         // Then
         XCTAssertEqual(analytics.receivedEvents.first, WooAnalyticsStat.orderProductDiscountRemove.rawValue)
     }
+
+    func test_calculatePriceAfterDiscount_when_discount_has_thousands_separators_calculates_price_correctly() {
+        // Given
+        let viewModel = FeeOrDiscountLineDetailsViewModel(isExistingLine: false,
+                                                          baseAmountForPercentage: 0,
+                                                          initialTotal: .zero,
+                                                          lineType: .fee,
+                                                          locale: usLocale,
+                                                          storeCurrencySettings: usStoreSettings,
+                                                          didSelectSave: { _ in })
+        viewModel.updateAmount("1,000.00")
+
+        // When
+        let discountedPrice = viewModel.calculatePriceAfterDiscount("1050.00")
```

Comment:
Not necessarily something to fix or update now, but I'm having a hard time wrapping my head about this particular method, I understand what are we testing as a whole, but I don't think the method definition or name is quite right: 

Passing a number like 1050 or other as parameter to `calculatePriceAfterDiscount` feels a bit arbitrary, since the function uses the discount the merchant might or might not already have entered to calculate the output internally, which makes us rely on calling other methods first like `updateAmount` in order to make it work, so it relies on side-effects to actually calculate something.

Perhaps we should add discount as an additional parameter and pass it in as well, so is clearer what's calculating over which values?
```
- func calculatePriceAfterDiscount(_ price: String) -> String { ... 
+ func stringFormattedPriceAfterDiscount(_ totalPrice: String, discount: Decimal) -> String { ... 
```

Reply:
I don't want to change that much about how the code works in this PR – I agree that the existing interface is strange, but this fix is already getting quite sprawling without changing the interface.

It's strange to say "calculate price after discount", passing in the _price_, rather than the discount. But passing the discount to the discount view model would also be weird, because you'd first have to get that discount from the view model you're about to pass it back to.

Either the `FeeOrDiscountLineDetailsViewModel` should provide the discount to something else to do the calculation, or it should know the price already and vend the discounted price in a published property. Probably the former.

Whether it's worth improving or not really depends on whether we're going to use it or change it in future though – refactoring it for the sake of a clearer interface isn't a good use of time if we're not going to do those things.